scripts/run_model_in_rocm_docker.sh

```diff
@@ -0,0 +1,121 @@
+#!/bin/bash
+
+MODEL="BERT-Large"
+#IMAGE="rocm/tensorflow:latest"
+IMAGE="rocm/tensorflow:rocm3.1-tf1.15-dev"
+
+# Print a message
+echo "This script will run the $MODEL model in a ROCm container."
+echo "Below is what it will do:"
+echo "  1. Pull the latest ROCm docker image;"
+echo "  2. Prepare a sample traing data set in a temporary directory;"
+echo "  3. Train $MODEL inside the ROCm container;"
+echo "  4. Clean up the temoprary directory created and exit."
+echo "Please press any key to start, or ESC to exit."
+
+# Read a key press
+read -n 1 -s KEY
+if [ "$KEY" == $'\e' ] ; then
+  echo "Exit. Did nothing."
+  exit 0
+fi
+
+# Get the folders
+SCRIPTPATH=$(dirname $(realpath $0))
+CODE_DIR=$SCRIPTPATH/..
+CODE_DIR_INSIDE=/data/code
+
+# Pull the docker image
+echo 
+echo "=== Docker pulling image $IMAGE and start container ==="
+docker pull $IMAGE
+CTNRNAME=RunDocker
+echo -n "Is $CTNRNAME running? "
+docker inspect -f '{{.State.Running}}' $CTNRNAME
+if [ $? -eq 0 ]; then
+    echo -n "Container $CTNRNAME is running. Stopping first ... "
+    docker stop $CTNRNAME
+fi
+echo "Starting $CTNRNAME"
+docker run --name $CTNRNAME -it -d --rm --network=host --device=/dev/kfd --device=/dev/dri --ipc=host --shm-size 16G --group-add video --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -w $CODE_DIR_INSIDE -v $CODE_DIR:$CODE_DIR_INSIDE $IMAGE
+
+# Preparing temporary folder for training
+TRAIN_DIR_NAME=dashboard_train_dir
+TRAIN_DIR=$CODE_DIR/$TRAIN_DIR_NAME
+rm -rf $TRAIN_DIR
+mkdir -p $TRAIN_DIR
+TRAIN_DIR_INSIDE=$CODE_DIR_INSIDE/$TRAIN_DIR_NAME
+
+# Copy the configuration file and vocab file
+MODEL_CONFIG_DIR=$CODE_DIR/configs/bert_large
+cp $MODEL_CONFIG_DIR/vocab.txt $TRAIN_DIR/vocab.txt
+cp $MODEL_CONFIG_DIR/bert_config.json $TRAIN_DIR/bert_config.json
+
+# Preparing training data
+echo 
+echo "=== Preparing training data ==="
+
+# Generate the training data
+DATA_SOURCE_FILE_PATH=sample_text.txt
+DATA_SOURCE_NAME=$(basename "$DATA_SOURCE_FILE_PATH")
+
+SEQ=128
+DATA_TFRECORD=${DATA_SOURCE_NAME}_seq${SEQ}.tfrecord
+
+# calculate max prediction per seq
+MASKED_LM_PROB=0.15
+calc_max_pred() {
+  echo $(python3 -c "import math; print(math.ceil($SEQ*$MASKED_LM_PROB))")
+}
+MAX_PREDICTION_PER_SEQ=$(calc_max_pred)
+
+if [ ! -f "$DATA_TFRECORD" ]; then
+  # generate tfrecord of data
+  docker exec $CTNRNAME \
+  python3 $CODE_DIR_INSIDE/create_pretraining_data.py \
+    --input_file=$CODE_DIR_INSIDE/$DATA_SOURCE_FILE_PATH \
+    --output_file=$TRAIN_DIR_INSIDE/$DATA_TFRECORD \
+    --vocab_file=$TRAIN_DIR_INSIDE/vocab.txt \
+    --do_lower_case=True \
+    --max_seq_length=$SEQ \
+    --max_predictions_per_seq=$MAX_PREDICTION_PER_SEQ \
+    --masked_lm_prob=$MASKED_LM_PROB \
+    --random_seed=12345 \
+    --dupe_factor=5
+fi
+
+# Perform training
+echo 
+echo "=== Training BERT ==="
+CUR_TRAINING=${DATA_SOURCE_NAME}_ba${BATCH}_seq${SEQ}
+mkdir -p $TRAIN_DIR/$CUR_TRAINING
+
+TRAIN_STEPS=200
+WARMUP_STEPS=50
+BATCH=8
+LEARNING_RATE=2e-5
```

Comment:
Any reason why this learning rate?

Reply:
It makes no difference if we are running only 200 steps. This is just to demonstrate that the code runs on ROCm.